orientdb/src/main/java/site/ycsb/db/OrientDBClient.java

```diff
@@ -1,272 +1,364 @@
-
-                db.create();
-              }
-            }
-          } else {
-            OServerAdmin server = new OServerAdmin(url).connect(user, password);
-
-            if (remoteStorageType == null) {
-              throw new DBException(
-                  "When connecting to a remote OrientDB instance, "
-                      + "specify a database storage type (plocal or memory) with "
-                      + STORAGE_TYPE_PROPERTY);
-            }
-
-            if (newdb) {
-              if (server.existsDatabase()) {
-                LOG.info("OrientDB drop and recreate fresh db");
-
-                server.dropDatabase(remoteStorageType);
-              }
-
-              server.createDatabase(db.getName(), ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
-            } else {
-              if (!server.existsDatabase()) {
-
-                LOG.info("OrientDB database not found, creating fresh db");
-                server.createDatabase(server.getURL(), ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
-              }
-            }
-
-            server.close();
-          }
-
-          dbChecked = true;
+        orient.createIfNotExists(dbName, dbType);
+        if (!orient.isOpen()) {
+          orient.open(dbName, cp.getUser(), cp.getPassword());
         }
-
-        if (db.isClosed()) {
-          db.open(user, password);
+        if (pool == null) {
+          pool = new ODatabasePool(orient, dbName, cp.getUser(), cp.getPassword());
         }
-
-        if (!db.getMetadata().getSchema().existsClass(CLASS)) {
-          db.getMetadata().getSchema().createClass(CLASS);
-        }
-
-        db.close();
-
-        if (databasePool == null) {
-          databasePool = new OPartitionedDatabasePool(url, user, password);
+        try (final ODatabaseSession session = pool.acquire()) {
+          final OClass newClass = session.createClassIfNotExist(CLASS);
+          LOG.info("OrientDB class created = " + CLASS);
+          newClass.createProperty("key", OType.STRING);
+          LOG.info("OrientDB class property created = 'key'.");
         }
-
+        createIndexForCollection(dbName);
         initialized = true;
+        LOG.info("OrientDB successfully initialized.");
       }
-    } catch (Exception e) {
+    } catch (final Exception e) {
       LOG.error("Could not initialize OrientDB connection pool for Loader: " + e.toString());
       e.printStackTrace();
     } finally {
-      INIT_LOCK.unlock();
+      REENTRANT_INIT_LOCK.unlock();
     }
+  }
 
+  private void initAndGetDatabaseUrlAndDbType(String url) {
+    if (url.startsWith("remote:")) {
+      final OrientDBConfigBuilder poolCfg = OrientDBConfig.builder();
+      poolCfg.addConfig(OGlobalConfiguration.DB_POOL_MIN, 5);
+      poolCfg.addConfig(OGlobalConfiguration.DB_POOL_MAX, 10);
+      final OrientDBConfig oriendDBconfig = poolCfg.build();
+      if (orient == null) {
+        orient = new OrientDB(url, "root", "admin", oriendDBconfig);
```

Comment:
This user/password should come from the ConnectionProperties we made before.

Reply:
done